projectLive/R/mod_study_summary.R

```diff
@@ -215,88 +219,54 @@ mod_study_summary_server <- function(
   output$data_focus_plot <- plotly::renderPlotly({
     
     shiny::req(
-      merged_dataset(), 
-      selected_study_name(),
-      input$data_focus_columns
+      filtered_merged_table(), 
+      input$data_focus_columns,
+      data_config
     )
     
-    column_list <- c(
-      "assay" = "Assays Used",
-      "resourceType" = "Resource Added",
-      "species" = "Species Used",
-      "tumorType" = "Tumor Types Investigated"
-    )
-
-    columns <- input$data_focus_columns
-    count_columns <- column_list[input$data_focus_columns]
+    param_list <- data_config %>% 
+      purrr::pluck(
+        "modules", 
+        "study_summary", 
+        "outputs", 
+        "data_focus"
+      ) 
     
-    data <- merged_dataset() %>%
-      dplyr::select(c("studyName", columns)) %>% 
-      dplyr::filter(studyName == selected_study_name()) %>% 
-      tidyr::pivot_longer(-"studyName") %>% 
-      tidyr::drop_na() %>% 
-      dplyr::group_by(.data$studyName, .data$name, .data$value) %>% 
-      dplyr::summarise("count" = dplyr::n()) 
+    data_list <- filtered_merged_table() %>% 
+      format_plot_data_with_param_list(param_list) %>% 
+      create_data_focus_tables(param_list$plot$x, input$data_focus_columns)
     
-    dfs <- purrr::map2(
-      columns,
-      count_columns,
-      create_plot_df_from_count_df,
-      data
-    )
-
-    plots <- purrr::pmap(
-      dplyr::tibble(
-        "data" = dfs,
-        "x" = "studyName",
-        "y" = count_columns,
-        "color" = columns
-      ) %>%
-        dplyr::mutate("fill" = .data$color),
-      create_study_summary_plot
-    )
-
-    plotly::subplot(plots, titleX = TRUE)
+    validate(need(length(data_list) > 0 , param_list$empty_table_message))
     
+    create_data_focus_plots(data_list, param_list, input$data_focus_columns)
   })
   
-  output$study_timeline <- plotly::renderPlotly({
-    shiny::req(merged_dataset(), selected_study_name())
-    data <- merged_dataset() %>%
-      dplyr::filter(
-        .data$studyName == selected_study_name(),
-        !is.na(.data$resourceType)
-      ) %>% 
-      dplyr::select(
-        "Study Name" = "studyName", 
-        "Resource Type" = "resourceType", 
-        "Year" = "year", 
-        "Month" = "month"
+  output$study_timeline_plot <- plotly::renderPlotly({
```

Comment:
This plot also needs a message similar to this
>validate(need(length(data_list) > 0 , param_list$empty_table_message))

Reply:
@jaybee84 Both plots in mod_study_summary.R have empty table messages correct?

Reply:
Study Timeline is not showing the empty table error message. I am seeing this:
>Error: ncol(Lab) == 3L || length(Lab) == 3 is not TRUE

Reply:
This was an issue when some of the tables in the subplot are empty and others are not. I've added code to fix this.